jsonp-policy/src/main/java/io/apiman/plugins/jsonp_policy/JsonpPolicy.java

```diff
@@ -76,8 +86,12 @@ public void write(IApimanBuffer chunk) {
                     if (firstChunk) {
                         IApimanBuffer buffer = bufferFactory.
                                 createBuffer(callbackFunctionName.length() + OPEN_PARENTHESES.length());
-                        buffer.append(callbackFunctionName);
-                        buffer.append(OPEN_PARENTHESES);
+                        try {
+                            buffer.append(callbackFunctionName, encoding);
+                            buffer.append(OPEN_PARENTHESES, encoding);
+                        } catch (UnsupportedEncodingException e) {
+                            throw new RuntimeException(e);
```

Comment:
You should capture these exceptions and call a failure (i.e. chain.doFailure() ), unless @EricWittmann thinks we should be doing that automatically in the chain. I seem to recall we did that before but there were some problems with it.